### PR TITLE
Fix Megatron-LM tokenizer path

### PR DESCRIPTION
**Pull request by hikari-kubota-fixstars**
# What does this PR do?
This PR addresses an issue with the Megatron-LM checkpoint conversion script introduced in the [documentation](https://huggingface.co/docs/accelerate/usage_guides/megatron_lm#utility-for-checkpoint-reshaping-and-interoperability). The path to the Megatron-LM tokenizer was outdated and did not match the current path. I have updated the script to reflect the correct path to the tokenizer.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@muellerzr @ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
cool! I think we need to check versions of megatron for that no?

**Comment by hikari-kubota-fixstars** [comment]
@ArthurZucker 

Thank you for pointing that out. Yes, you're correct. 



In version `0.5.0` (tag: `core_v0.5.0`), Megatron-LM was using:

```
from megatron.tokenizer.tokenizer import _vocab_size_with_padding
```

However, starting with version `0.6.0` (tag: `core_v0.6.0`), this was changed to:

```
from megatron.training.tokenizer.tokenizer import _vocab_size_with_padding
```

**Comment by hikari-kubota-fixstars** [comment]
@ArthurZucker 

 I’ve implemented the version checks for Megatron-LM and updated the code to handle different versions accordingly. If there are any additional changes or issues, please let me know!

**Comment by hikari-kubota-fixstars** [comment]
@ArthurZucker 

I'd like to add that the commit where the change from `megatron/tokenizer/tokenizer.py` to`  megatron/training/tokenizer/tokenizer.py` happened is here :
https://github.com/NVIDIA/Megatron-LM/commit/38644dd756c46142787b1bd5dc08ed82ca0e6de1#diff-4d0a12b13825799ff228dad8740d28ea3a79dd0e83cd5fb72f796be6a6e5ce03

File: src/transformers/models/megatron_gpt2/checkpoint_reshaping_and_interoperability.py [hunk 1]
```diff
@@ -607,7 +607,14 @@ def convert_checkpoint_from_transformers_to_megatron(args):
...
-        from megatron.tokenizer.tokenizer import _vocab_size_with_padding
+        from megatron.core import package_info
+
+        version = package_info.__version__
+        version = tuple(map(int, version.split(".")))
+        if version < (0, 6):
+            from megatron.tokenizer.tokenizer import _vocab_size_with_padding
+        else:
+            from megatron.training.tokenizer.tokenizer import _vocab_size_with_padding
     except ModuleNotFoundError:
```
Comment (ArthurZucker) [inline src/transformers/models/megatron_gpt2/checkpoint_reshaping_and_interoperability.py:618]: Thanks, you can use the utils: https://github.com/huggingface/transformers/blob/main/src/transformers/utils/import_utils.py#L42

**Review by ArthurZucker — COMMENTED** [review]
Thanks, this is better!

File: src/transformers/models/megatron_gpt2/checkpoint_reshaping_and_interoperability.py (hunk 1 above)
Comment (hikari-kubota-fixstars) [reply src/transformers/models/megatron_gpt2/checkpoint_reshaping_and_interoperability.py]: @ArthurZucker 

Megatron-LM is not a distributed package, so I couldn't use "https://github.com/huggingface/transformers/blob/main/src/transformers/utils/import_utils.py#L42".
 Therefore, I modified the code to use importlib.util to check the importability of Megatron-LM.

**Review by ArthurZucker — COMMENTED** [review]
Thanks for the update!

**Comment by hikari-kubota-fixstars** [comment]
@ArthurZucker 
Thank you for providing the sample code for 'check the version.' I have modified the code to use from packaging import version. If you have any other suggestions, please let me know.

**Review by ArthurZucker — APPROVED** [review]
Thanks

**Pull request merged**